ext/mysql2/extconf.rb

```diff
@@ -62,6 +62,14 @@ def asplode(lib)
   abort "-----\nCannot execute mysql_config at #{mc}\n-----" unless File.executable?(mc)
   warn "-----\nUsing mysql_config at #{mc}\n-----"
   ver = `#{mc} --version`.chomp.to_f
+  major, minor, inc = `#{mc} --version`.chomp.split('.').map( &:to_i )
+  if major > 5 || major == 5 && minor == 7 && inc >= 11
+    $CFLAGS << ' -D SSL_MODE_SUPPORT'
+  elsif major == 5 && minor == 7 && inc >= 3 && inc < 11
+    $CFLAGS << ' -D SSL_TOGGLE_SUPPORT'
+  else
+    $CFLAGS << ' -D NO_SSL_MODE_SUPPORT'
+  end
```

Comment:
mkmf func | enum member result  (ie. SSL_MODE_PREFRRED ) | enum result (ie mysql_ssl_mode)
-----------|---------------------------|------
have_const | true | false
have_func | true | false
have_header | true | false
have_library | false | false
have_macro | false | false
have_struct_member | false | n/a 
have_type | false | false
have_var | false | false

After hve_type failed, I tried everything.  Yay, learning about mkmf.  I think have_const is the winner.